### ISSUE TITLE
"does not derive from BaseException" for generic bound by `BaseException`

### ISSUE THREAD

**Issue by lmmarsano** [microsoft/pyright#1560]
**Describe the bug**
Type error for raising exceptions typed from generics bound by `BaseException`.

**To Reproduce**
Save the provided code as `stub.py` & run the CLI.

**Expected behavior**
No type error.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
import typing as T
_X = T.TypeVar('_X', bound=BaseException)
def raise_value(value: _X):
    raise value
```

**VS Code extension or command-line**
```shellsession
$ npx pyright stub.py --outputjson
```
```json
{
    "version": "1.1.117",
    "time": "1614737043879",
    "diagnostics": [
        {
            "file": "/home/LOM0227/project/ALM.IntakePage/stub.py",
            "severity": "error",
            "message": "Invalid exception class or object\n  \"_X@raise_value\" does not derive from BaseException",
            "range": {
                "start": {
                    "line": 3,
                    "character": 10
                },
                "end": {
                    "line": 3,
                    "character": 15
                }
            },
            "rule": "reportGeneralTypeIssues"
        },
        {
            "file": "/home/LOM0227/project/ALM.IntakePage/stub.py",
            "severity": "warning",
            "message": "TypeVar \"_X\" appears only once in generic function signature",
            "range": {
                "start": {
                    "line": 2,
                    "character": 23
                },
                "end": {
                    "line": 2,
                    "character": 25
                }
            },
            "rule": "reportInvalidTypeVarUse"
        }
    ],
    "summary": {
        "filesAnalyzed": 14,
        "errorCount": 1,
        "warningCount": 1,
        "informationCount": 0,
        "timeInSec": 0.264
    }
}
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next version.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.118, which I just published. It will also be included in the next release of Pylance.
